### ISSUE TITLE
Creation of Raw data files in integration test needs improvement

### ISSUE THREAD

**Issue by henrykironde** [weecology/retriever#575]
In the [integration test](https://github.com/weecology/retriever/blob/master/test/test_integration.py), we convert text to a raw dataset file

**for example:**  Given the data below,

```
simple_csv = {'name': 'simple_csv',
              'raw_data': "a,b,c\n1,2,3\n4,5,6\n",
              'script': "shortname: simple_csv\ntable: simple_csv, http://example.com/simple_csv.txt",
              'expect_out': 'a,b,c\n1,2,3\n4,5,6\n'}
```

we create a file; `simple_csv` and the raw_data is written to this file as : 

```
a,b,c
1,2,3
4,5,6
```

At times this does not happen as expected. Extra `endofline` characters are added causing wrong data when bulk inserting. 
An easy way out is by setting `*do_not_bulk_insert: True` in the script

**Comment by goelakash** [comment]
@henrykironde Why do extra `endofline` characters get added?

**Comment by henrykironde** [comment]
I think its from the way we write the raw data: `'raw_data': "a,b,c\n1,2,3\n4,5,6\n`
I have not looked at it, but I think we need to add some preprocessing in the `create_file` function.
Something like split string on "\n" and with each item write to file.

**Comment by ethanwhite** [comment]
What do the two of you think to switching over the something like:

`'raw_data': ["a,b,c", "1,2,3", "4,5,6"]`

and then having `create_file` do something like

`testfile.writelines(line + '\n' for line in data)`

This makes the raw data easier to read and probably decreases the chance of typos. For long lines it also allows to use formatting like:

```
'raw_data': ["a,b,c", 
             "1,2,3",
             "4,5,6"]
```

**Comment by goelakash** [comment]
I agree.

**Comment by henrykironde** [comment]
Am thinking of closing this issue, I think we may not need this anymore

**Comment by ethanwhite** [comment]
I still like the change proposed above since I think it's more readable and probably less error prone, but I don't think it's a high priority. Seems like something that could have an `easy` or `getting-started` tag attached to it and leave it for a new(ish) contributor to try to tackled at some point.

**Comment by ethanwhite** [comment]
Implemented in 3ec57441477253a5cfba51f41c50c6df231839c5 and c7d82726890c7d9c119cebb3a2e270c3230481ad. Thanks @henrykironde!
